### PR TITLE
[CWS-1348] Persist profile when a workload stops

### PR DESCRIPTION
**Pull request by YoannGh**
### What does this PR do?

This PR makes it so that the profile associated to a workload is persisted to the filesystem when the corresponding workload stops.
Upon loading profiles from the filesystem, persisted profiles are prioritized over profiles originating from activity dumps using a newly introduced selector field.

### CODE REVIEW

File: pkg/security/security_profile/profile/selector/selector.go [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux
+
+// Package selector holds selector related files
+package selector
```
Comment (spikat) [inline pkg/security/security_profile/profile/selector/selector.go:9]: Do we need this empty file?

File: pkg/security/security_profile/profile/selector/selector_proto_dec_v1.go [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+func ProtoToWorkloadSelector(selector *proto.ProfileSelector) cgroupModel.WorkloadSelector {
+	if selector == nil {
+		return cgroupModel.WorkloadSelector{}
+	}
+
+	return cgroupModel.WorkloadSelector{
+		Image: selector.GetImageName(),
+		Tag:   selector.GetImageTag(),
+	}
+}
```
Comment (spikat) [inline pkg/security/security_profile/profile/selector/selector_proto_dec_v1.go:27]: Same as encoder func: I wonder if we should rather put those functions in `resolvers/cgroup/` dir  (where the workload selector stands)?

File: pkg/security/security_profile/profile/selector/selector.go (hunk 1 above)
Comment (YoannGh) [reply pkg/security/security_profile/profile/selector/selector.go]: removed it!

**Review by spikat — APPROVED** [review]
LGTM

**Pull request merged**